### PR TITLE
B 19997 int

### PR DESCRIPTION
**Pull request by CoryKleinjanCACI**
## Summary

This task creates the standalone shipment functionality for the office app, and the UI for the TIO role. When a standalone shipment is part of a payment request, it will be displayed with a '- Standalone' tag on it to alert the TIO. Additionally, when they are reviewing the payment request it will be noted there that the shipment is a standalone, and in the calculation details it will display the Standalone Payment Cap, which is currently 1000$. Any Standalone Domestic Crating service item should not exceed this cap regardless of how large the crate is.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [x] Has the branch been pulled in and checked out?
- [x] Have the BL acceptance criteria been met for this change?
- [x] Was the CircleCI build successful?
- [x] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Setup: Run db_dev_fresh to ensure db changes are in place

Note: When testing I found that many of the service items for existing data are not complete, and will cause the creation of a payment request to error out. To test a payment request with multiple service items it is easiest to just create them yourself.

1. Access the Office app
2. Login to the Prime Sim
3. Add a domestic crating service item with the standalone box checked.

4. Login as a TOO
5. Find the Move you edited in the Prime Sim
6. Approve the newly created Service Item

5. Login to the Prime Sim
6. Navigate to the previously edited move
7. Create a payment request that includes the Standalone crating item you created

8. Login as a TIO
9. Find the move you edited in the Prime Sim
10. Navigate to the payment requests tab
11. Note that the Domestic crate service item you created has '- Standalone' after domestic crating
12. Review the payment request
13. Note that the Domestic crate service item you created has '- Standalone' after domestic crating in the payment request review
14. Click on calculations to view the calculations
15. Note that the standalone payment cap is displayed

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-05-31 at 10 03 46 AM

Screenshot 2024-05-31 at 10 20 10 AM

### CODE REVIEW

**Review by antgmann — COMMENTED** [review]
image
image
image
Looks good from the testing instructions! Though in hindsight, I suppose I should make a larger crate.

File: src/components/Office/ServiceItemCalculations/helpers.js [hunk 1]
```diff
@@ -616,6 +616,21 @@ const cratingSize = (params, mtoParams) => {
...
+  const label = SERVICE_ITEM_CALCULATION_LABELS.StandaloneCrate;
+
+  if (standalone === 'true') {
+    const value = getParamValue(SERVICE_ITEM_PARAM_KEYS.StandaloneCrateCap, params);
+    return calculation(value, label);
+  }
+
+  return calculation(0, label);
+};
+
```
Comment (antgmann) [inline src/components/Office/ServiceItemCalculations/helpers.js:633]: I'm getting a frontend console warning. Could it be caused by changes on this PR, or is it likely already existing? 
image

Comment (CoryKleinjanCACI) [reply src/components/Office/ServiceItemCalculations/helpers.js:633]: Let me check.

Comment (CoryKleinjanCACI) [reply src/components/Office/ServiceItemCalculations/helpers.js:633]: I checked on main branch. This console error was pre-existing.

**Review by antgmann — APPROVED** [review]

**Comment by r-mettler** [comment]
image
image

And because I am a 🤓 I am putting the screenshot with the item size I chose. Thats right, I am shipping PI in a standalone crate 🤓 
image

**Comment by r-mettler** [comment]
Everything looks good. Just had the question about whether the page should be displaying standalone cap in cents. If that is the case maybe we want to look at adding an indicator to let the users know its displayed in cents. Like a cents symbol after the amount or a label that says (in cents) or something. Otherwise it just looks weird and confusing.

**Pull request merged**